specification/marketplace/resource-manager/Microsoft.Marketplace/stable/2020-12-01/Marketplace.json

```diff
@@ -0,0 +1,1068 @@
+      "description": "Plan with requesters details",
+      "properties": {
+        "planId": {
+          "description": "Gets the plan id",
+          "type": "string",
+          "readOnly": true
+        },
+        "planDisplayName": {
+          "description": "Gets the plan display name",
+          "type": "string",
+          "readOnly": true
+        },
+        "requesters": {
+          "description": "Gets requesters details list",
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "$ref": "#/definitions/UserRequestDetails"
+          }
+        }
+      }
+    },
+    "UserRequestDetails": {
+      "description": "user request details",
+      "type": "object",
+      "properties": {
+        "user": {
+          "description": "Gets user id",
+          "type": "string",
+          "readOnly": true
+        },
+        "date": {
+          "description": "Gets request date",
+          "type": "string",
+          "readOnly": true
+        },
+        "justification": {
+          "description": "Gets justification",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "RequestApprovalsList": {
+      "description": "List of admin request approval resources",
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/RequestApprovalResource"
+          }
+        },
+        "nextLink": {
+          "description": "URL to get the next set of notifications list results if there are any.",
+          "type": "string"
+        }
+      }
+    },
+    "RequestApprovalResource": {
+      "description": "Request approval resource.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/RequestApprovalProperties",
+          "description": "The privateStore approval request data structure.",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "RequestApprovalProperties": {
+      "description": "Approval request resource properties",
+      "type": "object",
+      "properties": {
+        "offerId": {
+          "description": "Gets or sets unique offer id.",
+          "type": "string"
+        },
+        "offerDisplayName": {
+          "description": "Gets or sets offer display name",
+          "type": "string",
+          "readOnly": true
+        },
+        "publisherId": {
+          "description": "The offer's publisher id",
+          "type": "string"
+        },
+        "plansDetails": {
+          "description": "Gets or sets the plans details",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/planDetails"
+          }
+        },
+        "isClosed": {
+          "description": "Gets or sets a value indicating whether the request is closed",
```

Comment:
This property is marked as read only but the description says "gets or sets" for read only properties a customer cannot set any value, please review all your properties mark them as readOnly: true (if your service is setting them) and update the description accordingly.

Reply:
Forgot to mention, as a best practice, definitions' properties used only in GET APIs should be marked as readOnly, an exception to this is when a definition is also used in a PUT API (and even here, be careful on only marking as readOnly the properties that your service sets.)